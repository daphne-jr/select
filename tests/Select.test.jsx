import focusTest from './shared/focusTest';
import blurTest from './shared/blurTest';

describe('Select', () => {
  focusTest('single', {});
  blurTest('single');
});
