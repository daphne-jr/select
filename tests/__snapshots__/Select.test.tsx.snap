// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select render renders aria-attributes correctly 1`] = `
<div
  aria-label="some-label"
  aria-labelledby="test-id"
  class="antd select-test antd-single antd-allow-clear antd-show-arrow"
>
  <div
    class="antd-selector"
  >
    <span
      class="antd-selection-search"
    >
      <input
        aria-activedescendant="undefined_list_0"
        aria-autocomplete="list"
        aria-controls="undefined_list"
        aria-haspopup="listbox"
        aria-owns="undefined_list"
        autocomplete="off"
        class="antd-selection-search-input"
        role="combobox"
        style="opacity:0"
        value=""
      />
    </span>
    <span
      class="antd-selection-item"
    >
      2
    </span>
  </div>
  <span
    aria-hidden="true"
    class="antd-selection-arrow-icon"
    style="user-select:none;-webkit-user-select:none"
    unselectable="on"
  >
    <span
      class="antd-selection-arrow-icon-icon"
    />
  </span>
  <span
    aria-hidden="true"
    class="antd-selection-clear-icon"
    style="user-select:none;-webkit-user-select:none"
    unselectable="on"
  >
    <span
      class="antd-selection-clear-icon-icon"
    >
      ×
    </span>
  </span>
</div>
`;

exports[`Select render renders correctly 1`] = `
<div
  class="antd select-test antd-single antd-allow-clear antd-show-arrow"
>
  <div
    class="antd-selector"
  >
    <span
      class="antd-selection-search"
    >
      <input
        aria-activedescendant="undefined_list_0"
        aria-autocomplete="list"
        aria-controls="undefined_list"
        aria-haspopup="listbox"
        aria-owns="undefined_list"
        autocomplete="off"
        class="antd-selection-search-input"
        role="combobox"
        style="opacity:0"
        value=""
      />
    </span>
    <span
      class="antd-selection-item"
    >
      2
    </span>
  </div>
  <span
    aria-hidden="true"
    class="antd-selection-arrow-icon"
    style="user-select:none;-webkit-user-select:none"
    unselectable="on"
  >
    <span
      class="antd-selection-arrow-icon-icon"
    />
  </span>
  <span
    aria-hidden="true"
    class="antd-selection-clear-icon"
    style="user-select:none;-webkit-user-select:none"
    unselectable="on"
  >
    <span
      class="antd-selection-clear-icon-icon"
    >
      ×
    </span>
  </span>
</div>
`;

exports[`Select render renders data-attributes correctly 1`] = `
<div
  class="antd select-test antd-single antd-allow-clear antd-show-arrow"
  data-id="12345"
  data-test="test-id"
>
  <div
    class="antd-selector"
  >
    <span
      class="antd-selection-search"
    >
      <input
        aria-activedescendant="undefined_list_0"
        aria-autocomplete="list"
        aria-controls="undefined_list"
        aria-haspopup="listbox"
        aria-owns="undefined_list"
        autocomplete="off"
        class="antd-selection-search-input"
        role="combobox"
        style="opacity:0"
        value=""
      />
    </span>
    <span
      class="antd-selection-item"
    >
      2
    </span>
  </div>
  <span
    aria-hidden="true"
    class="antd-selection-arrow-icon"
    style="user-select:none;-webkit-user-select:none"
    unselectable="on"
  >
    <span
      class="antd-selection-arrow-icon-icon"
    />
  </span>
  <span
    aria-hidden="true"
    class="antd-selection-clear-icon"
    style="user-select:none;-webkit-user-select:none"
    unselectable="on"
  >
    <span
      class="antd-selection-clear-icon-icon"
    >
      ×
    </span>
  </span>
</div>
`;

exports[`Select render renders disabled select correctly 1`] = `
<div
  class="antd select-test antd-single antd-allow-clear antd-show-arrow antd-disabled"
>
  <div
    class="antd-selector"
  >
    <span
      class="antd-selection-search"
    >
      <input
        aria-activedescendant="undefined_list_0"
        aria-autocomplete="list"
        aria-controls="undefined_list"
        aria-haspopup="listbox"
        aria-owns="undefined_list"
        autocomplete="off"
        class="antd-selection-search-input"
        disabled=""
        role="combobox"
        style="opacity:0"
        value=""
      />
    </span>
    <span
      class="antd-selection-item"
    >
      2
    </span>
  </div>
  <span
    aria-hidden="true"
    class="antd-selection-arrow-icon"
    style="user-select:none;-webkit-user-select:none"
    unselectable="on"
  >
    <span
      class="antd-selection-arrow-icon-icon"
    />
  </span>
  <span
    aria-hidden="true"
    class="antd-selection-clear-icon"
    style="user-select:none;-webkit-user-select:none"
    unselectable="on"
  >
    <span
      class="antd-selection-clear-icon-icon"
    >
      ×
    </span>
  </span>
</div>
`;

exports[`Select render renders dropdown correctly 1`] = `
<div
  class="antd select-test antd-single antd-allow-clear antd-show-arrow antd-open"
>
  <div
    class="antd-selector"
  >
    <span
      class="antd-selection-search"
    >
      <input
        aria-activedescendant="undefined_list_0"
        aria-autocomplete="list"
        aria-controls="undefined_list"
        aria-haspopup="listbox"
        aria-owns="undefined_list"
        autocomplete="off"
        class="antd-selection-search-input"
        role="combobox"
        value=""
      />
    </span>
    <span
      class="antd-selection-item"
    >
      2
    </span>
  </div>
  <div>
    <div
      class="antd-dropdown  antd-dropdown-placement-bottomLeft "
    >
      <div>
        <div
          id="undefined_list"
          role="listbox"
          style="height:0;overflow:hidden"
        >
          <div
            aria-selected="false"
            id="undefined_list_0"
            role="option"
          />
          <div
            aria-selected="false"
            id="undefined_list_1"
            role="option"
          >
            jack
          </div>
          <div
            aria-selected="false"
            id="undefined_list_2"
            role="option"
          >
            lucy
          </div>
        </div>
        <div
          class=""
          style="height:200px"
        >
          <div>
            <div
              class=""
              style="display:flex;flex-direction:column"
            >
              <div
                class="antd-item antd-item-group"
              >
                manager
              </div>
              <div
                aria-selected="false"
                class="antd-item antd-item-option antd-item-option-grouped antd-item-option-active"
              >
                <b
                  style="color:red"
                >
                  jack
                </b>
                <span
                  aria-hidden="true"
                  class="antd-item-option-selected-icon"
                  style="user-select:none;-webkit-user-select:none"
                  unselectable="on"
                >
                  <span
                    class="antd-item-option-selected-icon-icon"
                  >
                    ✓
                  </span>
                </span>
              </div>
              <div
                aria-selected="false"
                class="antd-item antd-item-option antd-item-option-grouped"
              >
                lucy
                <span
                  aria-hidden="true"
                  class="antd-item-option-selected-icon"
                  style="user-select:none;-webkit-user-select:none"
                  unselectable="on"
                >
                  <span
                    class="antd-item-option-selected-icon-icon"
                  >
                    ✓
                  </span>
                </span>
              </div>
              <div
                class="antd-item antd-item-group"
              >
                engineer
              </div>
              <div
                aria-selected="false"
                class="antd-item antd-item-option antd-item-option-grouped"
              >
                yiminghe
                <span
                  aria-hidden="true"
                  class="antd-item-option-selected-icon"
                  style="user-select:none;-webkit-user-select:none"
                  unselectable="on"
                >
                  <span
                    class="antd-item-option-selected-icon-icon"
                  >
                    ✓
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    aria-hidden="true"
    class="antd-selection-arrow-icon"
    style="user-select:none;-webkit-user-select:none"
    unselectable="on"
  >
    <span
      class="antd-selection-arrow-icon-icon"
    />
  </span>
  <span
    aria-hidden="true"
    class="antd-selection-clear-icon"
    style="user-select:none;-webkit-user-select:none"
    unselectable="on"
  >
    <span
      class="antd-selection-clear-icon-icon"
    >
      ×
    </span>
  </span>
</div>
`;

exports[`Select render renders role prop correctly 1`] = `
<div
  class="antd select-test antd-single antd-allow-clear antd-show-arrow"
  role="button"
>
  <div
    class="antd-selector"
  >
    <span
      class="antd-selection-search"
    >
      <input
        aria-activedescendant="undefined_list_0"
        aria-autocomplete="list"
        aria-controls="undefined_list"
        aria-haspopup="listbox"
        aria-owns="undefined_list"
        autocomplete="off"
        class="antd-selection-search-input"
        role="combobox"
        style="opacity:0"
        value=""
      />
    </span>
    <span
      class="antd-selection-item"
    >
      2
    </span>
  </div>
  <span
    aria-hidden="true"
    class="antd-selection-arrow-icon"
    style="user-select:none;-webkit-user-select:none"
    unselectable="on"
  >
    <span
      class="antd-selection-arrow-icon-icon"
    />
  </span>
  <span
    aria-hidden="true"
    class="antd-selection-clear-icon"
    style="user-select:none;-webkit-user-select:none"
    unselectable="on"
  >
    <span
      class="antd-selection-clear-icon-icon"
    >
      ×
    </span>
  </span>
</div>
`;
